### PR TITLE
docs: fix `ng lint` and tools lint command in documentation 

### PR DESCRIPTION
**Pull request by tmair**
**Description:**
This PR tries to fix the `ng lint` and the tools `lint` command of the documentation site (see also https://github.com/ReactiveX/rxjs/pull/7364#pullrequestreview-1720976136).
It is quite a substantial change, as introducing `eslint` into the build requires to change also changes in some of the source or tests files. 

**Related issue (if exists):**
This is trying to address the request for fixing `ng lint` sufaced in PR https://github.com/ReactiveX/rxjs/pull/7364#pullrequestreview-1720976136

### CODE REVIEW

**Comment by benlesh** [comment]
@jakovljevic-mladen or @niklas-wortmann can you please take a look at this one?

**Comment by benlesh** [comment]
ping @jakovljevic-mladen or @niklas-wortmann

**Comment by niklas-wortmann** [comment]
Hey sorry for the delay, I am going to have a look at this after the holidays. Happy holidays everyone

**Review by jakovljevic-mladen — COMMENTED** [review]
Hey @tmair, thanks a lot for this PR.

I have left couple of comments that we could address, most of them being related to turning off `no-non-null-assertion`, but also some other small issues that we could fix.

File: apps/rxjs.dev/src/app/app.component.ts [hunk 1]
```diff
@@ -230,23 +230,24 @@ export class AppComponent implements OnInit {
...
     combineLatest(
       this.navigationService.versionInfo,
-      this.navigationService.navigationViews.pipe(map((views) => views['docVersions']))
+      this.navigationService.navigationViews.pipe(map((views) => views.docVersions))
     ).subscribe(([versionInfo, versions]) => {
       this.docVersions = [...versions];
 
       // Find the current version - either title matches the current deployment mode
       // or its title matches the major version of the current version info
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/src/app/app.component.ts]: Yes, I removed them and reverted the changes that where made for it.

File: apps/rxjs.dev/angular.json [hunk 2]
```diff
@@ -1,20 +1,25 @@
 {
-  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "$schema": "../../node_modules/@angular/cli/lib/config/schema.json",
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/angular.json:2]: It seems that the hoist algorithm moves the `@angular/cli` package to the root of the workspace.

File: apps/rxjs.dev/src/app/custom-elements/code/code.component.ts [hunk 3]
```diff
@@ -139,7 +140,7 @@ export class CodeComponent implements OnChanges {
       .pipe(tap(() => this.codeFormatted.emit()))
       .subscribe(
         (c) => this.setCodeHtml(c),
-        (err) => {
+        (_err) => {
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/src/app/custom-elements/code/code.component.ts]: Done

File: apps/rxjs.dev/src/app/custom-elements/resource/resource.service.spec.ts [hunk 4]
```diff
@@ -88,62 +88,62 @@ describe('ResourceService', () => {
 function getTestResources() {
   // tslint:disable:quotemark
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/src/app/custom-elements/resource/resource.service.spec.ts]: Done

File: apps/rxjs.dev/src/app/shared/select/select.component.spec.ts [hunk 5]
```diff
@@ -47,15 +47,15 @@ describe('SelectComponent', () => {
       fixture.detectChanges();
       const span = getButton().querySelector('span');
       expect(span).not.toEqual(null);
-      expect(span!.className).toContain('symbol');
+      expect(span?.className).toContain('symbol');
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/src/app/shared/select/select.component.spec.ts]: Done

File: apps/rxjs.dev/src/app/shared/toc.service.ts [hunk 6]
```diff
@@ -57,14 +57,15 @@ export class TocService {
     const div: HTMLDivElement = this.document.createElement('div');
     div.innerHTML = heading.innerHTML;
     const anchorLinks: NodeListOf<HTMLAnchorElement> = div.querySelectorAll('a');
+    // eslint-disable-next-line @typescript-eslint/prefer-for-of
```
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/src/app/shared/toc.service.ts]: Yes, that is possible.
Done.

**Comment by tmair** [comment]
I rebased the PR against master and addressed the review comments

File: apps/rxjs.dev/src/app/shared/toc.service.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (tmair) [reply apps/rxjs.dev/src/app/shared/toc.service.ts:56]: Nevermind, NodeLists do not implement the iterator protocol. So `for of` loops are not possible for that use case.

**Comment by tmair** [comment]
One more question: Should we add the lint command to the build pipeline or should that be a separate PR once this PR is merged?

File: apps/rxjs.dev/src/app/shared/toc.service.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (demensky) [reply apps/rxjs.dev/src/app/shared/toc.service.ts:56]: I think the problem is the missing [dom.iterable](https://github.com/microsoft/TypeScript/blob/0ea57f6ca19c23e0bc690bd2706f6b6e11a97e02/src/lib/dom.iterable.d.ts#L44) in `tsconfig.json`.

[the remark this answers is not in the document]
Comment (jakovljevic-mladen) [reply apps/rxjs.dev/src/app/shared/toc.service.ts:56]: Not a big deal.

**Review by jakovljevic-mladen — APPROVED** [review]
LGTM, thanks a lot again @tmair.

**Comment by jakovljevic-mladen** [comment]
> Should we add the lint command to the build pipeline or should that be a separate PR once this PR is merged?

You can do that in the next PR.

**Pull request merged**